Tests/GRPCCodeGenTests/Internal/Translator/ServerCodeTranslatorSnippetBasedTests.swift

```diff
@@ -0,0 +1,398 @@
+/*
+ * Copyright 2023, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import XCTest
+
+@testable import GRPCCodeGen
+
+extension SnippetBasedTranslatorTests {
+  func testServerCodeTranslatorUnaryMethod() throws {
+    let method = MethodDescriptor(
+      documentation: "Documentation for unaryMethod",
+      name: "unaryMethod",
+      isInputStreaming: false,
+      isOutputStreaming: false,
+      inputType: "NamespaceA_ServiceARequest",
+      outputType: "NamespaceA_ServiceAResponse"
+    )
+    let service = ServiceDescriptor(
+      documentation: "Documentation for ServiceA",
+      name: "ServiceA",
+      namespace: "namespaceA",
+      methods: [method]
+    )
+    let expectedSwift =
+      """
+      protocol namespaceA.ServiceA.StreamingServiceProtocol: RegistrableRPCService, Sendable {
+          func unaryMethod(request: ServerRequest.Stream<namespaceA.ServiceA.Methods.unaryMethod.Input>) async throws -> ServerResponse.Stream<namespaceA.ServiceA.Methods.unaryMethod.Output>
+      }
+      // Generated conformance to `RegistrableRPCService`.
+      public extension namespaceA.ServiceA.StreamingServiceProtocol {
+          func registerRPCs(with router: inout RPCRouter) {
+              router.registerHandler(
+                  for: namespaceA.ServiceA.Methods.unaryMethod.descriptor,
+                  deserializer: ProtobufDeserializer<namespaceA.ServiceA.Methods.unaryMethod.Input>(),
+                  serializer: ProtobufSerializer<namespaceA.ServiceA.Methods.unaryMethod.Output>(),
+                  handler: { request in
+                      try await self.unaryMethod(request)
+                  }
+              )
+          }
+      }
+      protocol namespaceA.ServiceA.ServiceProtocol: namespaceA.ServiceA.StreamingServiceProtocol {
+          func unaryMethod(request: ServerRequest.Single<namespaceA.ServiceA.Methods.unaryMethod.Input>) async throws -> ServerResponse.Single<namespaceA.ServiceA.Methods.unaryMethod.Output>
+      }
+      // Generated partial conformance to `namespaceA.ServiceA.StreamingServiceProtocol`.
+      public extension namespaceA.ServiceA.ServiceProtocol {
+          func unaryMethod() {
```

Comment:
This isn't right: it should match the signature of the method in the streaming protocol